web/src/navigate.ts

```diff
@@ -10,9 +10,34 @@ function go_to_row(msg_id: number): void {
     message_lists.current.select_id(msg_id, {then_scroll: true, from_scroll: true});
 }
 
+function is_long_message(message_height: number): boolean {
+    // Here we just need to the message to be long enough
+    // to be hard to completely see just using `up / down`
+    // to warrant use of `page up / down`.
+    // Also, we want to avoid triggering `page up / down`
+    // when pressing `up / down` on short messages to avoid
+    // the selected message being barely visible.
+    return message_height >= 0.5 * message_viewport.height();
```

Comment:
I think we can a bit lenient on the long message offset here to allow more messages to be passed as a long message.